### ISSUE TITLE
`quarkus.http.cors=true` does not work on `main`

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#35374]
### Describe the bug

`BearerTokenAuthorizationInGraalITCase` started failing from #35246, replacing `quarkus.http.cors=true` with the new `quarkus.http.cors.enabled=true` fixes the problem in #35373.

According to https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.4 `quarkus.http.cors=true`  is still supported which is a nice idea to have it supported for a while, but unfortunately it does not work in native.

Roberto, @radcortez, have a look please when you get a chance, I believe there is some time before 3.4.0 is released. Thanks

**Comment by sberyozkin** [comment]
This is not specific to `native` - this property is set in OIDC tests but it is not really useful for the actual tests, for some reasons it was set a long time ago, probably assuming HttpUnit would be enforcing it.

I've just noticed in a not-native mode:

```
[WARNING] [io.quarkus.config] Unrecognized configuration key "quarkus.http.cors" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
```

For some reasons it is also causing the test failure in native

**Comment by sberyozkin** [comment]
Guillaume, I've just set a milestone so that this issue is fixed when Roberto gets a chance before 3.4.0, otherwise we'll a get a ton of issues immediately after 3.4.0 release :-)

**Comment by radcortez** [comment]
I'll have a look this week. 

Sorry just came back from PTO and had more urgent issues to look at.
